project/src/com/google/daggerquery/example/models/Apartment.java

```diff
@@ -0,0 +1,40 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ */
+
+package com.google.daggerquery.example.models;
+
+import javax.inject.Inject;
+
+public class Apartment {
+  @Inject
+  Bed bed;
+
+  @Inject
+  MiniBar miniBar;
+
+  @Inject
+  TV tv;
+
+  @Inject
+  Phone phone;
+
+  @Inject
+  Tourist tourist;
+
+  @Inject
```

Comment:
You should pick either "constructor injection" or "field injection".

In the case of constructor injection, the constructor is marked with `@Inject`, but not the fields:

```
final MiniBar miniBar;
...

@Inject
Apartment(MiniBar miniBar, ...) {
  this.miniBar = minibar;
  ...
}
```

In the case of field injection, fields are marked with `@Inject`, but not the constructor:
```
@Inject MiniBar miniBar;
...

Apartment() {}
```

Constructor injection is typically preferred, because it allows the fields to be marked `final`.

Reply:
Thank you! I have removed all _field injection_ cases and made fields `private final`.